### PR TITLE
[Rule] Implement too-long-variable-name

### PR DESCRIPTION
**Pull request by p-zander**
Implements #1443

New rule checks all variable names in contexts such as

- `*** Variables ***` section (suite variables)
- `Set Variable` and `VAR` syntax (RF7+)
- `Set Local/Test/Task/Suite/Global Variable`
- Iteration variables used in `FOR`-loops
- Keyword `[Arguments]`-setting
- Keyword arguments [embedded into the keyword name](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#toc-entry-365) (BDD)
- Variables used for catching error messages in `TRY-EXCEPT AS` branches

A variable name that is longer than `too-long-variable-name.max_len (default: 40)` is reported **once per scope**, so an offending name is not overreported if it's assigned to several times.

Supports severity thresholds as well.

Additionally some reusable function were moved to utils/misc.py.

Need to complete test data for all edge cases; Might add some options to the rule if it makes sense.

### CODE REVIEW

File: src/robocop/linter/rules/lengths.py [hunk 1]
```diff
@@ -803,6 +844,104 @@ def count_keyword_calls(self, node):
...
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.variables: dict[str, tuple] = {}
+
+    def visit_Node(self, node) -> None:  # noqa: N802
+        if not node.errors:
+            self.generic_visit(node)
+
+    def visit_Scope(self, node: Keyword | TestCase | VariableSection):  # noqa: N802
```
Comment (bhirsz) [inline src/robocop/linter/rules/lengths.py:868]: Is visit_Scope actual visitor from Robot we're overwriting, or simply our function name? If so, we could just simply name it after one of the visitors in line 876 (like visit_Keyword)

File: src/robocop/linter/rules/lengths.py [hunk 2]
```diff
@@ -609,6 +622,27 @@ class EmptyKeywordTagsRule(Rule):
...
+
+    name = "too-long-variable-name"
+    rule_id = "LEN32"
+    message = "Variable name '{variable_name}' is too long ({variable_name_length}/{allowed_length})"
+    severity = RuleSeverity.WARNING
+    parameters = [
+        RuleParam(name="max_len", default=40, converter=int, desc="allowed length of a variable name"),
+    ]
+    severity_threshold = SeverityThreshold("max_len", compare_method="greater", substitute_value="allowed_length")
+    added_in_version = "6.6.x"
```
Comment (p-zander) [inline src/robocop/linter/rules/lengths.py:640]:
```suggestion
    added_in_version = "6.6.2"
```

**Comment by p-zander** [comment]
Still open points:

- One instance of underreporting I don't quite understand, but might be acceptable: [here](https://github.com/MarketSquare/robotframework-robocop/pull/1450/files#diff-fd03ea4e5bec9bc9cd8c3063109bc75723516d12c14df2fd446c38a5c07b9f21R35)
- Adding test data for type hints/conversion
- Adding invalid test data

Other than that I'm pretty happy with the results, although the implementation might look a bit wonky.

File: src/robocop/linter/rules/lengths.py [hunk 3]
```diff
@@ -609,6 +622,27 @@ class EmptyKeywordTagsRule(Rule):
...
+    sonar_qube_attrs = sonar_qube.SonarQubeAttributes(
+        clean_code=sonar_qube.CleanCodeAttribute.FOCUSED, issue_type=sonar_qube.SonarQubeIssueType.CODE_SMELL
```
[the remark this answers is not in the document]
Comment (bhirsz) [reply src/robocop/linter/rules/lengths.py:642]: I am also not sure! If you copied those from length rules, it should be fine. I tried to apply those attributes with accordance to what I have found on Sonar Qube documentation, but most of it reuses the same attributes in case of our rules anyway.

**Comment by p-zander** [comment]
Mh, btw how sure are you that in your workflows the tests are actually executed against all these different RF versions? At least one package is updated [here](https://github.com/MarketSquare/robotframework-robocop/actions/runs/17516421821/job/49754471309?pr=1450#step:6:14), and if I had to guess that would be `robotframework` itself.

If I try to do it locally (downgrade RF and then run `uv run` with `--verbose` flag) I see this 
image

**Comment by bhirsz** [comment]
I am quite sure if it comes for robocop pre merge. After merge..not so, since we also switch to uv and it could have its own quirks. Thanks for pointing it out, I will also take a look what's happening here.

**Comment by bhirsz** [comment]
> Mh, btw how sure are you that in your workflows the tests are actually executed against all these different RF versions? At least one package is updated [here](https://github.com/MarketSquare/robotframework-robocop/actions/runs/17516421821/job/49754471309?pr=1450#step:6:14), and if I had to guess that would be `robotframework` itself.
> 
> If I try to do it locally (downgrade RF and then run `uv run` with `--verbose` flag) I see this 
> image
> 

I am quite sure if it comes for robocop pre merge. After merge..not so, since we also switch to uv and it could have its own quirks. Thanks for pointing it out, I will also take a look what's happening here.

**Comment by p-zander** [comment]
Yeah, so I tested locally without `uv run` and tests were failing/should have been failing before for at least some earlier RF versions. I'm also not really familar with uv, but the `--with` option seems to be what you want: https://docs.astral.sh/uv/concepts/projects/run/#requesting-additional-dependencies

**Comment by p-zander** [comment]
I think I might be done with this for now. Please let me know, if I should add some options to the rule or anything.
